contracts/cw1155-base/src/contract.rs

```diff
@@ -513,7 +513,7 @@ fn query_tokens(
     limit: Option<u32>,
 ) -> StdResult<TokensResponse> {
     let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
-    let start = start_after.map(Bound::exclusive);
+    let start = start_after.as_ref().map(|s| Bound::exclusive(s.as_str()));
```

Comment:
Curious as to why `as_ref()` is needed.... ah, never mind lifetimes... we cannot consume the original `start_after` until `start` is completed as it maintains a reference there...

Reply:
It's because the key itself is a reference, so we need an `&Addr` for the bound here. This is a way to obtain a reference to something that is in scope, when it's inside an `Option`.